### PR TITLE
tests: Add integration test for http api

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve? 
#2412 

### What is changed and how it works?
Add integration tests for all Open API.

### Check List 

Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by asddongmen** [comment]
I triger `/run-all-tests` times to check if the added tests is stable.

File: tests/http_api/util/test_case.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
+import sys
+import requests as rq
+import time
+import json
+
+# http status code
+OK = 200
```
Comment (hi-rustin) [inline tests/http_api/util/test_case.py:7]: Could we use the existing http library?

File: tests/http_api/util/test_case.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    url = BASE_URL+"/changefeeds"
+    # create changefeed
+    for i in range(1, 4):
+        data = {
+            "changefeed_id": "changefeed-test"+str(i),
+            "sink_uri": "blackhole://",
+            "ignore_ineligible_table": True
+        }
+        # set sink_uri
+        if i == 1 and sink_uri != "":
```
Comment (hi-rustin) [inline tests/http_api/util/test_case.py:29]: Why do we need to determine if it is 1?

File: tests/http_api/util/test_case.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+    print("pass test: create changefeed")
+
+
+def list_changefeed():
+    # test state: all
+    url = BASE_URL+"/changefeeds?state=all"
+    resp = rq.get(url)
+    assert resp.status_code == OK
+    data = resp.json()
```
Comment (hi-rustin) [inline tests/http_api/util/test_case.py:55]: It seems we don't use this value?

File: tests/http_api/run.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    python $CUR/util/test_case.py get_status
+
+    python $CUR/util/test_case.py create_changefeed "$SINK_URI"
+
+    run_sql "CREATE table test.simple(id int primary key, val int);"
+    run_sql "CREATE table test.\`simple-dash\`(id int primary key, val int);"
+    # wait for above sql done in the up source
+    sleep 1
+
+    python $CUR/util/test_case.py list_changefeed
```
Comment (hi-rustin) [inline tests/http_api/run.sh:42]: It looks like these tests have sequential dependencies, maybe we need to add comments to suggest not to modify these sequences?

**Review by hi-rustin — APPROVED** [review]
rest LGTM 👍

File: tests/http_api/run.sh (hunk 4 above)
Comment (amyangfei) [reply tests/http_api/run.sh:42]: What about using an array for sequential cases, such as
```
sequential_cases=("list_changefeed" "get_changefeed" ...)
for case in $sequential_cases; do
    python $CUR/util/test_case.py $case
done
```

Comment (asddongmen) [reply tests/http_api/run.sh:42]: good idea ! thanks!

File: tests/http_api/util/test_case.py (hunk 3 above)
Comment (asddongmen) [reply tests/http_api/util/test_case.py]: done

File: tests/http_api/util/test_case.py (hunk 1 above)
Comment (asddongmen) [reply tests/http_api/util/test_case.py]: done

**Review by amyangfei — APPROVED** [review]

**Pull request merged**